api/v1/logistics.proto

```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
+
+package coopnorge.logistics.api.v1;
+
+import "google/api/annotations.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+// ---------------------------------------
+// Service
+// ---------------------------------------
+
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+    info: {
+        title: "Coop Norge Interview Assignment - Coop Logistics API"
+        version: "1.0.0"
+        description: "Coop Logistics Engine, simulates several cargo units that deliver cargo between different storages in the world in other words generating events."
+    }
+};
+
+// CoopLogisticsEngineAPI
+service CoopLogisticsEngineAPI {
+    // MoveUnit request will be send when unit moves in dimensions to new location.
+    rpc MoveUnit(MoveUnitRequest) returns (DefaultResponse) {
+        option (google.api.http) = {
+            post: "/v1/cargo_unit/move"
+            body: "*"
+        };
+    }
+    // UnitReachedWarehouse reports when unit reached warehouse to do something there.
+    rpc UnitReachedWarehouse(UnitReachedWarehouseRequest) returns (DefaultResponse) {
+        option (google.api.http) = {
+            post: "/v1/warehouse/cargo_unit/reached"
+            body: "*"
+        };
+    }
+}
+
+// ---------------------------------------
+// Requests
+// ---------------------------------------
+
+// MoveUnitRequest
+message MoveUnitRequest {
+    int64 cargo_unit_id = 1;
+    Location location = 2;
+}
+
+// UnitReachedWarehouseRequest contains WarehouseAnnouncement with Location
+message UnitReachedWarehouseRequest {
+    Location location = 1;
+    WarehouseAnnouncement announcement = 2;
+}
+
+// ---------------------------------------
+// Responses
+// ---------------------------------------
+
+// DefaultResponse
+message DefaultResponse {}
+
+// ---------------------------------------
+// Models
+// ---------------------------------------
+
+// WarehouseAnnouncement
+message WarehouseAnnouncement {
+    // cargo_unit_id is unique id
+    int64 cargo_unit_id = 1;
+    // warehouse_id is unique id
+    int64 warehouse_id = 2;
+    // the message contains information about the announcement
+    string message = 3;
+}
+
+// Location where entity now located in X,Y Axis
+message Location {
+    uint32 x = 1;
```

Comment:
Probably not that important as it's for an assessment, but does one not generally use latitude and longitude in logistics?

Reply:
Up to you, but if anything then just to avoid the inevitable "why no lat/long" that I would ask if I was doing the interview.